python/dlisio/plumbing/errorhandler.py

```diff
@@ -4,10 +4,124 @@
 from .. import core
 
 class ErrorHandler(core.error_handler):
-    """
-    TODO ErrorHandler
-    """
+    """ Defines rules about error handling
+
+    Many .dlis files happen to be not compliant with specification or simply
+    broken. This class gives user some control over handling of such files.
+
+    For example:
```

Comment:
Rather than listing some examples here, I would spell out _all_ the different categories (critical, major, minor, info) and what the generally mean, default behaviour and maybe an example. Without a definition lines like 14 "... dlisio doesn't consider critical..."  makes no sense. I read that and think: but what makes dlisio consider something to be critical??  

>When dlisio encounters a spec violation it categories the issue based on the severity of the violation. Some issues are easily ignorable while other might force dlisio to give up on its current task. `ErrorHandler` supplies an interface for changing how dlisio react to different violation in the file. The different categories are `info`, `minor`, `major` and `critical`:

E.g critical: 

> Any issue that forces dlisio stop its current objective prematurely is categorised as `critical`. By default a `critical` error raises a `RuntimeError`. An example would be file indexing, which happens at load. Suppose the indexing fails midways through the file. There is no way for dlisio to reliably keep indexing the file. However, it is likely that the file is readable up until the point of failure. Changing the behaviour of `critical` from raising an Exception to logging would in this case mean that a partially indexed file is returned by load.